csharp/extractor/Semmle.Extraction.CSharp.Standalone/DotNet.cs

```diff
@@ -1,17 +1,72 @@
-﻿using System.Diagnostics;
+﻿using System;
+using System.Diagnostics;
 
 namespace Semmle.BuildAnalyser
 {
     /// <summary>
     /// Utilities to run the "dotnet" command.
     /// </summary>
-    internal static class DotNet
+    internal class DotNet
     {
-        public static int RestoreToDirectory(string projectOrSolutionFile, string packageDirectory)
+        private readonly ProgressMonitor progressMonitor;
+
+        public DotNet(ProgressMonitor progressMonitor)
         {
-            using var proc = Process.Start("dotnet", $"restore --no-dependencies \"{projectOrSolutionFile}\" --packages \"{packageDirectory}\" /p:DisableImplicitNuGetFallbackFolder=true");
+            this.progressMonitor = progressMonitor;
+            Info();
+        }
+
+        private void Info()
+        {
+            try
+            {
+                // TODO: make sure the below `dotnet` version is matching the one specified in global.json
+                progressMonitor.RunningProcess("dotnet --info");
+                using var proc = Process.Start("dotnet", "--info");
+                proc.WaitForExit();
+                var ret = proc.ExitCode;
+                if (ret != 0)
+                {
+                    progressMonitor.CommandFailed("dotnet", "--info", ret);
+                    throw new Exception($"dotnet --info failed with exit code {ret}.");
+                }
+            }
+            catch (Exception ex)
```

Comment:
The purpose of this try/catch block is to catch exceptions throws by the process, but is also catches the exception that is thrown in l. 31 (and wraps it). Would it make sense to re-factor slightly?

Reply:
Huhh, good point.

Reply:
Excellent!